pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -309,6 +354,12 @@ func (a *ApplicationBackupController) handle(ctx context.Context, backup *stork_
 	if backup.Status.Stage == stork_api.ApplicationBackupStageFinal {
 		return nil
 	}
+
+	// Initialize execRulesCompleted for the backup UID
```

Comment:
This is redundant code. You already checking ifExists..

Reply:
That is in finalizer here it is during initialization.